modin/engines/ray/utils.py

```diff
@@ -164,30 +133,19 @@ def initialize_ray(
 """,
             )
             object_store_memory = Memory.get()
-            plasma_directory = RayPlasmaDir.get()
```

Comment:
why do we no longer need a configuration for plasma directory?..

also if we don't need it then we should probably remove `RayPlasmaDir` configuration variable

Reply:
`_plasma_directory` is not public anymore. Setting `_plasma_directory` automatically disables default object spilling (out of core).

If users want a custom setup for Ray, they should initialize themselves. The defaults here should make sense for most users and if someone wants to tweak things they can initialize Ray themselves. I am in favor of simplifying the configurations and initialization in the interest of simplifying the code. If the users need to set 3 or 4 configs they might as well just initialize Ray themselves.

Reply:
Well, less knobs for us (and users) is always better if this doesn't hurt.

In this case we should probably get rid of `RayPlasmaDir` configuration variable in `modin.config` altogether along with removing it here.

Reply:
Done!

Reply:
Since as of Ray 1.3+ object spilling is a default option in Ray shouldn't we update our documentation related to out of core feature?

Reply:
Yes, and we should probably disable the config/flag.

Reply:
When user initializes Ray on their own, they may miss important settings that Modin does for Ray initialization. For example, if Ray is initialized before Modin, then Ray doesn't get `OMP_NUM_THREADS=1` and all workers spawn as many threads as there are CPU cores leading to oversubscription and sometimes process gets killed for using too many resources.
Maybe we should at least document it somewhere that initializing Ray before Modin requires special handling.

Reply:
Ray also sets `OMP_NUM_THREADS=1` by default. I think Modin started doing it before Ray because the contention was noticeable in Modin very early on.

https://docs.ray.io/en/master/serve/core-apis.html?highlight=omp_num_threads#configuring-parallelism-with-omp-num-threads

Are there other best practices we should include in documentation @gshimansky?

Reply:
I wasn't aware that Ray sets OMP_NUM_THREADS now. It is something new to me. I don't have any other objections to ask user to initialize Ray manually.